boards/intel/intel_adsp/Kconfig.intel_adsp

```
@@ -0,0 +1,18 @@
+# Copyright (c) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_INTEL_ADSP_CAVS25
+	bool
+	select SOC_INTEL_CAVS_V25
+
+config BOARD_INTEL_ADSP_CAVS25_TGPH
+	bool
+	select SOC_INTEL_CAVS_V25
+
+config BOARD_INTEL_ADSP_ACE15_MTPM
+	bool
+	select SOC_INTEL_ACE15_MTPM
+
+config BOARD_INTEL_ADSP_ACE20_LNL
+	bool
+	select SOC_INTEL_ACE20_LNL
```

Please, no bool, as the `BOARD_<name>` and `BOARD_<full_identifier>` are constructed here:
https://github.com/zephyrproject-rtos/zephyr/blob/db742d03aa8fa469a96435fe85b292e62d061d38/boards/Kconfig.v2#L8
https://github.com/zephyrproject-rtos/zephyr/blob/db742d03aa8fa469a96435fe85b292e62d061d38/boards/Kconfig.v2#L13

Some details can be found here: https://docs.zephyrproject.org/latest/build/kconfig/tips.html#common-kconfig-shorthands

Please do:
```suggestion
config BOARD_INTEL_ADSP
	select SOC_INTEL_CAVS_V25 if BOARD_INTEL_ADSP_CAVS25
	select SOC_INTEL_CAVS_V25 if BOARD_INTEL_ADSP_CAVS25_TGPH
	select SOC_INTEL_ACE15_MTPM if BOARD_INTEL_ADSP_ACE15_MTPM
	select SOC_INTEL_ACE20_LNL if  BOARD_INTEL_ADSP_ACE20_LNL
```
as that will ensure any typos or similar wrong selection is caught by CI.

and it's fewer lines 😉

---

> Please, no bool, as the `BOARD_<name>` and `BOARD_<full_identifier>` are constructed here:

well, as I recall, without these `bool` it fails on build as config variables has no type for other boards except the current one.
@tejlmand m.b. there is some other problem with this way of board definition change then ?

---

> and it's fewer lines 😉

right, that's it, thank you, changed.